backend/src/main/java/ba/unsa/etf/si/bbqms/ws/controllers/GroupController.java

```diff
@@ -0,0 +1,157 @@
+package ba.unsa.etf.si.bbqms.ws.controllers;
+
+import ba.unsa.etf.si.bbqms.auth_service.api.AuthService;
+import ba.unsa.etf.si.bbqms.domain.BranchGroup;
+import ba.unsa.etf.si.bbqms.admin_service.api.GroupService;
+import ba.unsa.etf.si.bbqms.domain.User;
+import ba.unsa.etf.si.bbqms.ws.models.*;
+import org.springframework.http.ResponseEntity;
+import org.springframework.security.access.prepost.PreAuthorize;
+import org.springframework.web.bind.annotation.*;
+
+import java.util.List;
+import java.util.stream.Collectors;
+
+@RestController
+@RequestMapping("/api/v1/groups")
+public class GroupController {
+    private final GroupService groupService;
+    private final AuthService authService;
+
+    public GroupController(final GroupService groupService, AuthService authService) {
+        this.groupService = groupService;
+        this.authService = authService;
+    }
+
+    @GetMapping("/{tenantCode}")
+    @PreAuthorize("hasAnyRole('ROLE_SUPER_ADMIN', 'ROLE_BRANCH_ADMIN')")
+    public ResponseEntity getAll(@PathVariable final String tenantCode){
+        final User currentUser = authService.getAuthenticatedUser();
+
+        if(!currentUser.getTenant().getCode().equals(tenantCode)) {
+            return ResponseEntity.badRequest().build();
+        }
+
+        try {
+            List<BranchGroup> branchGroupList = groupService.getAll();
+            List<BranchGroupResponseDto> branchGroupResponseDtoList = branchGroupList.stream()
+                    .map(BranchGroupResponseDto::fromEntity).collect(Collectors.toList());
+            return ResponseEntity.ok().body(branchGroupResponseDtoList);
+        } catch (Exception e) {
+            return ResponseEntity.badRequest().body(new ErrorResponseDto(e.getMessage()));
+        }
+    }
+
+    @PostMapping("/{tenantCode}")
+    @PreAuthorize("hasAnyRole('ROLE_SUPER_ADMIN', 'ROLE_BRANCH_ADMIN')")
+    public ResponseEntity addBranchGroup(@PathVariable final String tenantCode, @RequestBody final BranchGroupCreateDto request) {
+        final User currentUser = authService.getAuthenticatedUser();
+
+        if(!currentUser.getTenant().getCode().equals(tenantCode)) {
+            return ResponseEntity.badRequest().build();
+        }
+
+        try {
+            return ResponseEntity.ok().body(BranchGroupResponseDto.fromEntity(groupService.addBranchGroup(request)));
```

Comment:
Can we reformat this a bit? Kind of hard to read as it is now. Maybe extract `groupService.addBranchGroup(request)` to a variable.